### PR TITLE
Diagnostics

### LINKED ISSUE #250: Diagnostics
We should perhaps remove the somewhat arbitrary `Warning` for covariance collapse, and replace it with some more sensible diagnostics. 

### (Initial) Suggestions
- At start: Configuration (inflation/localization/EKP method etc.)
- During run: Loss error, ensemble failures, timestep, current covariance determinant,

### PR DESCRIPTION
**Pull request by eviatarbach**
Closes #250

### CODE REVIEW

File: src/EnsembleKalmanSampler.jl [hunk 1]
```diff
@@ -143,4 +152,11 @@ function update_ensemble!(
     # but stored in data container with N_ens as the 2nd dim
 
     compute_error!(ekp)
+
+    # Diagnostics
+    cov_new = cov(get_u_final(ekp), dims = 2)
```
Comment (odunbar) [inline src/EnsembleKalmanSampler.jl:157]: `get_u_cov_final(ekp)`

File: src/UnscentedKalmanInversion.jl [hunk 2]
```diff
@@ -666,6 +683,11 @@ function update_ensemble!(
 
     push!(uki.u, DataContainer(u_p, data_are_columns = true))
 
+    if uki.verbose
+        cov_new = cov(u_p, dims = 2)
```
Comment (odunbar) [inline src/UnscentedKalmanInversion.jl:687]: `get_u_cov_final(uki)`

File: src/UnscentedKalmanInversion.jl [hunk 3]
```diff
@@ -652,6 +658,17 @@ function update_ensemble!(
 
     u_p_old = get_u_final(uki)
 
+    if uki.verbose
+        cov_init = cov(u_p_old, dims = 2)
```
Comment (odunbar) [inline src/UnscentedKalmanInversion.jl:662]: get_u_cov_final(uki)

File: src/EnsembleKalmanProcess.jl [hunk 4]
```diff
@@ -127,6 +131,7 @@ function EnsembleKalmanProcess(
     rng::AbstractRNG = Random.GLOBAL_RNG,
     failure_handler_method::FM = IgnoreFailures(),
     localization_method::LM = NoLocalization(),
+    verbose::Bool = true,
```
Comment (odunbar) [inline src/EnsembleKalmanProcess.jl:134]: Is it common to have this as `false` by default? (as in theory people will only need it when checking convergence / debugging?

File: src/EnsembleKalmanSampler.jl [hunk 5]
```diff
@@ -124,8 +123,18 @@ function update_ensemble!(
     # u: N_ens × N_par
     # g: N_ens × N_obs
     u_old = get_u_final(ekp)
+    cov_init = cov(u_old, dims = 2)
```
Comment (odunbar) [inline src/EnsembleKalmanSampler.jl:126]: `get_cov_u_final(ekp)`

**Review by odunbar — APPROVED** [review]
Hi Eviatar, this looks great. 

My only comment is to use `get_cov` methods where we can, as we have written them for this purpose.

Also verbose flags are often `false` by default i think? But I'll leave this up to you. 

Please address these then LGTM for squash and merge!

**Comment by eviatarbach** [comment]
Fails codecov due to new code, but no need to add tests for the diagnostics.

**Pull request merged**